### DISCUSSION TITLE
How to view a new Extension version's README from an unsupported platform?

### DISCUSSION THREAD

**Discussion by ericdrobinson** [microsoft/vscode-discussions#32] [Extension Development QnA] [answered]
We just pushed a new version of the [ExtendScript Debugger](https://marketplace.visualstudio.com/items?itemName=Adobe.extendscript-debug) live. The v1 versions were marked as Universal because platform-specific options did not exist when they were published.

We have since pushed v2.0.0 live with a build marked to support Windows 32/64 and macOS on Intel. The problem is that we would very much like people to **stop** using the v1 versions as they have serious problems (including that they simply do not work on macOS on Apple Silicon/Windows on ARM with no reporting for the user - the v2 version has this).

Additionally, even though the v2.0.0 version is live, the Marketplace Webpage listing shows the README contents for v1.1.2. **PLEASE** tell me that this isn't because we're viewing the listing from a browser and this is somehow interpreted as being viewed from the `web` platform... That would be a ridiculous assumption. The Marketplace listing should show the latest version when viewed from a browser. It would be outrageous for it to assume that "viewed in a browser" somehow means that you're viewing it from "VS Code in the browser" - assuming that this is what's causing the v1.1.2 

Several questions:

1. Is there a way to deprecate older versions so that they do not show up on the website anymore?
   - I think it's fine that they remain downloadable in the Version History, but the README should show the latest version unless someone somehow specifically selects an older version to see.
2. Is there a way to retroactively update the supported platform information for older releases?
3. Can this situation be more clearly messaged (e.g. metadata reads "v2.0.0" but the README is from "v1.1.2"; no indication that there's a more recent version that is unsupported on my current platform; etc.)?
   - This is a pretty confusing situation, even for those of us who published the extension. It took me a little while to figure out that I was seeing the old README and **not** the new one because I was viewing the webpage from an unsupported platform.
4. Is there a way to at least _link_ to a specific version on the Marketplace so that that version's README can be viewed by people who are on an unsupported platform?

At this point, it feels a bit like a mistake that we opted to specify the supported versions with the update because people on unsupported platforms are only able to see the older, broken versions.

**Comment by alefragnani** [comment, marked as answer]
> Is there a way to deprecate older versions so that they do not show up on the website anymore?

Unfortunately not yet, but there is an open issue in VS Marketplace repo (https://github.com/microsoft/vsmarketplace/issues/235). 

Another approach would be publish a new version BUT, I totally understand (and agree) this is not a good alternative.

What I'm seeing is a _Marketplace webpage issue_, because the README content inside VS Code Desktop and vscode.dev shows the correct, updated v2 content.

Maybe the new README wasn't approved by the VS Marketplace validation process (I remember some changes occurred sometime ago and some extensions had issues with it), but it's still weird, because the content is updated inside VS Code 😕

**Reply by ericdrobinson** [reply]
We actually think this may be the Marketplace seeing browsers as the "unsupported `web` platform" and therefore showing the old Universal version details. If this is the case, it is a **serious** bug. There is no reason that the webpage listing shouldn't show the latest details to users viewing it in a browser and **not** in a VS Code environment context.

Regardless, we have "addressed" the issue by doing the following:

- Release `v2.0.1` as a Universal build.
- Release `v2.0.2` as a platform-specific build.

We now get what we expect (mostly): the webpage shows the v2 README (presumably for people on the browser they are seeing the `v2.0.1` README). The only issue is that unsupported platforms will now see the `v2.0.1` release as available for unsupported platforms.

The Marketplace **really** needs two things:

1. A way to retroactively adjust the supported platforms for prior releases.
2. Marketplace pages viewed from a browser should _**IGNORE**_ platform specific limitations and show the latest version details.

This is currently an annoying mess.
